core/state/dump.go

```diff
@@ -34,16 +37,76 @@ type DumpAccount struct {
 	Storage  map[string]string `json:"storage"`
 }
 
+// For output in a collected format, as one large map
 type Dump struct {
 	Root     string                 `json:"root"`
 	Accounts map[string]DumpAccount `json:"accounts"`
 }
 
-func (self *StateDB) RawDump() Dump {
-	dump := Dump{
-		Root:     fmt.Sprintf("%x", self.trie.Hash()),
+// DumpAccountFull is the same as DumpAccount but also with address, for standalone printing
+type DumpAccountFull struct {
+	Address  string            `json:"address"`
+	Balance  string            `json:"balance"`
+	Nonce    uint64            `json:"nonce"`
+	Root     string            `json:"root"`
+	CodeHash string            `json:"codeHash"`
+	Code     string            `json:"code"`
+	Storage  map[string]string `json:"storage"`
+}
+
+// For line-by-line json output
+type IterativeDump struct {
+	encoder *json.Encoder
+}
+
+// Collector interface which the state trie calls during iteration
+type collector interface {
+	onRoot(common.Hash)
+	onAccount(string, DumpAccount)
+}
+
+func newCollectingDump() *Dump {
+	return &Dump{
 		Accounts: make(map[string]DumpAccount),
 	}
+}
```

Comment:
Given that we only ever use this method once, I don't think it's worth the namespace pollution of adding one extra method. Lets just inline it.